src/app/[locale]/dashboard/@promptForm/actions.tsx

```
@@ -47,6 +51,12 @@ export async function submit(
   error?: { message: string };
   board?: BoardRecord;
 }> {
+  const session = await getServerSession(authConfig);
+
+  if (!session) {
+    return { error: { message: 'Failed to get session, please log in' } }; //redirect to login
```

Tech debt  issue created 
#110

---

Are you sure we need to add this validation here? The middleware should be handling this by default, do you have a way to reproduce this error?

---

You are right! I will remove it

---

_In contrast to useSession, which will return a session object whether or not a user has logged in (whether or not cookies are present), getServerSession only returns a session object when a user has logged in (only when authenticated cookies are present), otherwise, it returns null._ from [getServerSession next oAuth docs](https://next-auth.js.org/configuration/nextjs#in-app-router)

---

You can throw an exception, unless you want to capture this specific use case in the UI. But I don't think that's going to happen as this might not be reproducible.
Do you have a way to reproduce the session being `null` there?

---

No, I couldn't reproduce it. Probably never get it. So you recommend thrown an exception instead?

---

Yeah, this is more of a runtime error. If it happens we should expect the server to fail. We can create an error page (if it ever happens) for the user